packages/block-editor/src/components/media-upload/check.js

```diff
@@ -1,23 +1,17 @@
 /**
  * WordPress dependencies
  */
-import { withSelect } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
 
-export function MediaUploadCheck( {
-	hasUploadPermissions,
-	fallback = null,
-	children,
-} ) {
+export function MediaUploadCheck( { fallback = null, children } ) {
+	const hasUploadPermissions = useSelect( ( select ) => {
+		const { getSettings } = select( 'core/block-editor' );
+		return !! getSettings().mediaUpload;
+	} );
```

Comment:
This also should have a dependency array:
```suggestion
	}, [] );
```